TM1py/Services/ElementService.py

```diff
@@ -350,19 +350,20 @@ def get_elements_filtered_by_attribute(self, dimension_name: str, hierarchy_name
         :param attribute_value:
         :return: List of element names
         """
-        attribute_name = attribute_name.replace(" ", "")
-        if isinstance(attribute_value, str):
-            url = format_url("/api/v1/Dimensions('{}')/Hierarchies('{}')?$expand=Elements($filter = "
-                             "Attributes/{} eq '{}';$select=Name)",
-                             dimension_name, hierarchy_name, attribute_name, attribute_value)
 
+        if isinstance(attribute_value, str):
+            mdx = (
+                f'{{FILTER({{TM1SUBSETALL([{dimension_name}].[{hierarchy_name}])}},'
+                f'[{dimension_name}].[{hierarchy_name}].[{attribute_name}] = "{attribute_value}")}}'
+            )
         else:
-            url = format_url("/api/v1/Dimensions('{}')/Hierarchies('{}')?$expand=Elements($filter = "
-                             "Attributes/{} eq {};$select=Name)",
-                             dimension_name, hierarchy_name, attribute_name, attribute_value)
+            mdx = (
+                f'{{FILTER({{TM1SUBSETALL([{dimension_name}].[{hierarchy_name}])}},'
+                f'[{dimension_name}].[{hierarchy_name}].[{attribute_name}] = {attribute_value})}}'
+            )
 
-        response = self._rest.GET(url, **kwargs)
-        return [elem['Name'] for elem in response.json()['Elements']]
+        elems =  self.execute_set_mdx(mdx)
```

Comment:
Since we are returning only a list of element names, I think it would make sense to explicitly only retrieve the element names in the `execute_set_mdx` function. That should speed up the execution. Kinda like this:

``` python
from TM1py import TM1Service

with TM1Service(address='localhost', port=12354, ssl=True, user="admin", password="apple") as tm1:
    mdx = "{Tm1SubsetAll([d1])}"
    elements = tm1.elements.execute_set_mdx(
        mdx=mdx,
        member_properties=["Name"],
        parent_properties=None,
        element_properties=None)

    print(elements)
```

Reply:
@MariusWirtz  updated as suggested.